src/main/java/com/iexec/core/chain/adapter/BlockchainAdapterService.java

```diff
@@ -167,4 +168,10 @@ public PublicChainConfig getPublicChainConfig() {
         }
         return null;
     }
+
+    @Bean
+    public int getChainId() {
+        return getPublicChainConfig().getChainId();
```

Comment:
Why not injecting the `PublicChainConfig`?
It could be done in the next PR btw.

Reply:
I'm often more on the side of only inject what is needed, but injecting PublicChainConfig can be done.